src/Akka.Hosting.TestKit/TestKit.cs

```diff
@@ -172,6 +159,11 @@ public async Task InitializeAsync()
 
             await _initialized.Task;
             
+            var system = _host.Services.GetRequiredService<ActorSystem>();
+            var registry = _host.Services.GetRequiredService<ActorRegistry>();
+            base.InitializeTest(system, (ActorSystemSetup)null!, null, null);
```

Comment:
By moving the call here we should be back in the normal `SynchronizationContext` the test runner uses, which should force the `TestActor` to always be the expected implicit sender, rather than randomly failing sometimes.